src/providers/openstack/configdrive.rs

```diff
@@ -144,17 +144,21 @@ impl OpenstackConfigDrive {
 impl MetadataProvider for OpenstackConfigDrive {
     fn attributes(&self) -> Result<HashMap<String, String>> {
         let mut out = HashMap::with_capacity(6);
-        let metadata_ec2: MetadataEc2JSON = self.read_metadata_ec2()?;
         let metadata_openstack: MetadataOpenstackJSON = self.read_metadata_openstack()?;
         if let Some(hostname) = metadata_openstack.hostname {
             out.insert("OPENSTACK_HOSTNAME".to_string(), hostname);
         }
-        if let Some(instance_id) = metadata_ec2.instance_id {
-            out.insert("OPENSTACK_INSTANCE_ID".to_string(), instance_id);
-        }
         if let Some(uuid) = metadata_openstack.uuid {
             out.insert("OPENSTACK_INSTANCE_UUID".to_string(), uuid);
         }
+
+        let metadata_ec2 = match self.read_metadata_ec2() {
+            Ok(metadata) => metadata,
+            Err(error) => return Ok(out),
```

Comment:
First, let's instead change `read_metadata_ec2()` to return a `Result<Option<MetadataEc2JSON>>`. So then it would return `Ok(None)` in this case. It should only return that if the error was `ENOENT`. Grep the codebase for `NotFound` for examples of this.

Second, this is a lot of metadata we're not injecting now. Are you sure that information is not available in the `openstack/meta_data.json`? I was under the impression when we talked about this that it was.

Reply:
All the metadata we are not injecting seems to be either ec2 related (INSTANCE_ID, INSTANCE_TYPE), or somewhat irrelevant, at least in my use case: IPV4_LOCAL and IPV4_PUBLIC .

The information provided by metal3 is the following:
```
{
  "local-hostname": "bmh-vm-02",
  "local_hostname": "bmh-vm-02",
  "metal3-name": "bmh-vm-02",
  "metal3-namespace": "test-capi",
  "name": "bmh-vm-02",
  "uuid": "30aed4c7-05a4-4d15-be9b-9194dc1b5ad1"
}
```

We can see uuid, hostname - although none of its label it's matching the current provider's key - and other metal3 related keys (useful to build the providerID pattern expected by metal3).